contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -281,26 +304,82 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         emit ProgramEnabled({ pool: pool, status: status, remainingRewards: p.remainingRewards });
     }
 
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function setAutoProcessRewardsCount(uint256 newAutoProcessRewardsCount)
+        external
+        greaterThanZero(newAutoProcessRewardsCount)
+        onlyAdmin
+    {
+        uint256 prevAutoProcessRewardsCount = _autoProcessRewardsCount;
+        if (prevAutoProcessRewardsCount == newAutoProcessRewardsCount) {
+            return;
+        }
+
+        _autoProcessRewardsCount = newAutoProcessRewardsCount;
+
+        emit AutoProcessRewardsCountUpdated({
+            prevAutoProcessRewardsCount: prevAutoProcessRewardsCount,
+            newAutoProcessRewardsCount: newAutoProcessRewardsCount
+        });
+    }
+
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function autoProcessRewards() external nonReentrant {
+        uint256 numOfPools = _pools.length();
+        uint256 index = _autoProcessRewardsIndex;
+        uint256 count = _autoProcessRewardsCount;
+        uint256 maxCount = Math.min(count * 2, numOfPools);
+
+        for (uint256 i = 0; i < maxCount; i++) {
+            bool completed = _processRewards(Token(_pools.at(index % numOfPools)), true);
+            index += 1;
+            if (completed) {
+                count -= 1;
+                if (count == 0) {
+                    break;
+                }
+            }
+        }
+
+        _autoProcessRewardsIndex = index;
+    }
+
     /**
      * @inheritdoc IAutoCompoundingRewards
      */
     function processRewards(Token pool) external nonReentrant {
+        _processRewards(pool, false);
+    }
+
+    /**
+     * @dev processes the rewards of a given pool
+     * returns true if the rewards processing was completed, and false if it was skipped
+     */
+    function _processRewards(Token pool, bool skipRecent) private returns (bool) {
         ProgramData memory p = _programs[pool];
 
         uint32 currTime = _time();
 
         if (!p.isEnabled || currTime < p.startTime) {
-            return;
+            return false;
+        }
+
+        if (skipRecent && currTime < p.prevDistributionTimestamp + AUTO_PROCESS_REWARDS_MIN_TIME_DELTA) {
+            return false;
         }
 
         uint256 tokenAmountToDistribute = _tokenAmountToDistribute(p, currTime);
         if (tokenAmountToDistribute == 0) {
-            return;
+            return false;
         }
 
         uint256 poolTokenAmountToBurn = _poolTokenAmountToBurn(pool, p, tokenAmountToDistribute);
         if (poolTokenAmountToBurn == 0) {
-            return;
+            return false;
```

Comment:
should return `true` here as well

Reply:
done

Reply:
No tests for the change?

Reply:
Hmmmmm... I'm not even sure that we have tests for the original, i.e., the tests generally just set up a bunch of programs with some returning false and some returning true (which up until now was returned only on the primary execution path of this function); I'm not sure that we have tests that go into each one of the preliminary 'return' statements.

Reply:
Done in the sense that the test covers the following execution paths:
- Return `false` from the 2nd `if` statement (skip due to minimum time period not elapsed)
- Return `true` from the 3rd `if` statement (amount of tokens to distribute is zero)
- Return `true` from the end of the function (tokens distributed and pool tokens burned)